setup.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import sys
+import sysconfig
+import platform
+import subprocess
+import pybind11
+
+from distutils.version import LooseVersion
+from setuptools import setup, Extension, find_packages
+from setuptools.command.build_ext import build_ext
+from setuptools.command.test import test as TestCommand
+from shutil import copyfile, copymode
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir=""):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+
+
+class CMakeBuild(build_ext):
```

Comment:
What are the advantages of using `cmake` here? Can't `setuptools` build an extension directly?

Reply:
I suppose we could, but we already have cmake baked into this repo and it was relatively simple to set it up to use cmake.   If you are aware of some pitfalls of using this type of workflow (setup.py -> cmake) I'm open to changing it, but I found this method relatively painleses.